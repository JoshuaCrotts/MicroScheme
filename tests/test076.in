(define zero? (λ (x) (= x 0)))
(define one? (λ (x) (= x 1)))
(define abs (λ (x) (if (<= x 0) (* -1 x) x)))

(define caddr (λ (x) (car (cdr (cdr x)))))

(zero? 0.01)
(zero? 1)
(zero? (- 0.01 0.01))
(zero? 0)
(one? 0)
(one? 1.01)
(one? (- 1.01 0.01))
(one? 1)
(abs -5)
(abs 0)
(abs 5)
(define cadr (λ (x) (car (cdr x))))
(cadr '(+ x y z q *))
(caddr '(+ x y z q *))
