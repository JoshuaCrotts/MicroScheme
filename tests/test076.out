#f
#f
#t
#t
#f
#f
#t
#t
5
0
5
x
y
